camel-core/src/main/java/org/apache/camel/ComponentVerifier.java

```diff
@@ -412,6 +411,12 @@ public String toString() {
         }
     }
 
+    class ExceptionAttributeImpl extends ErrorAttribute implements VerificationError.ExceptionAttribute {
+        private ExceptionAttributeImpl(String name) {
+            super(name);
+        }
+    }
+
```

Comment:
This is an initial shot for the move to enums (for one attribute). Looks good, the only drawback I see is that I have to introduce a temporary class, which can be instantiated but is still visible (since every inner class in an interface is public).

Two alternatives:

* Use always anonymous classes. But this blows up the code and also the types in the jars.
* Use an extra, external and package scoped class creating these objects (also for the custom ErrorCode and ErrorAttribute). This could be in the package o.a.c.i.verifier but would introduce again a cyclic (package) dependency.

The second approach is nicer, except when you insist of cycle free deps ;-)